blog/wdio-blog.md

```diff
@@ -0,0 +1,284 @@
+# Meta:
+- Title: React Native Application UI testing with WDIO
+
+- Keywords: React Native, iOS and Android UI testing, WebDriverIO, Appium, SauceLabs, Real Device testing
+
+- Summary: We at Godaddy, recently adopted WebDriverIO based UI testing for our React Native application thats in development phase. Benefits of using WDIO included that we could write UI tests similar to how we write tests for the web. Synchronous nature of WebDriverIO helps in writing clean UI tests along with PageObject models. WDIO config allows to plugin SauceLabs Emulators or Real Devices for cloud-based testing.
+
+- Author: [Raja Panidepu](https://github.com/rajapanidepu).
+
+# Article:
+
+## Motivation: 
+I work on a Mobile App building team which employs React Native Framework. Similar to how we write JavaScript code to develop ReactNative components for Android or iOS platform, we wanted to write our UI tests once and run on both the platforms. The answer we found is WebDriverIO + Appium.
+
+WebDriverIO is a Node.js implementation of WebDriver Protocol. WebDriverIO is the most popular Web UI testing framework, which also supports Native Mobile UI testing.
+
+The next part of the article discusses setting up WebDriverIO of the following scenarios:
+1. UI tests running locally using Emulators
+2. Using Cloud-based service to run on Emulators
+3. Using Cloud-based service to run on Real Devices
+4. Timeouts at various levels
+
+## Part 1: Setup WDIO to run tests native UI tests locally.
+Let's install WebDriverIO globally and create a basic WebDriverIO config.
+
+```console
+$ npm install webdriverio --save-dev
+$ wdio config
+```
+ 
+![Screenshot showing wdio config](wdio-config.png)
+
+Follow the prompts to create base WDIO configuration as shown in the above picture.
+
+Note: For now, set the base URL to the default value: http://localhost, we will update that later. Once done, let WebDriverIO install all the needed packages.
+
+Finally, below WebDriverIO config would be saved in `wdio.conf` file:
```

Comment:
"Finally, save the below WebDriverIO config in the `wdio.conf` file"

Reply:
This is not something that the user saves. Rather is the output of `wdio config` command.